### DISCUSSION TITLE
Error: unable to generate parameter set

### DISCUSSION THREAD

**Discussion by samwise-commits** [getporter/porter#1443] [Q&A]
When trying to run a 'porter parameters' to generate some params, I get the following error:

Error: unable to generate parameter set: Unexpected Escape Sequence: ['\x1b' 'O']

If I try to use my arrow keys to change the param type it just errors out to the above. I am on MacOSX running porter v0.33.0 (991a5a23).

Anyone know what is going on?

Thanks!

**Comment by carolynvs** [comment]
Let me repeat back what is happening to make sure I've got it:

1. You run `porter parameters generate --reference getporter/porter-hello:v0.1.0`
2. When you use the arrow keys to select where to get the value from, e.g. env var, secret, file, etc instead of moving the selection up/down instead it prints those characters and the command immediately fails.

Let me know if I missed a step, maybe you had to press enter before the command failed, or it works fine for porter-hello but not for your bundle, etc. 

Which shell are you using, bash, zsh?

**Reply by samwise-commits** [reply]
Hello! Thank you for the reply back!

Correct, I run that command, except for my bundle, and as soon as I use the up or down arrow keys to select the value I get an immediate error about the escape sequence.

I also tried running your porter-hello bundle but it just automatically sets 1 parameter and doesn't let me select anything.

I've tried using bash, zsh, and even powershell. Fails in all.
Thanks!

**Reply by samwise-commits** [reply]
Okay not entirely sure what happened... messed around with the quickstart guide to create the hello world bundle, and I can now select parameters. I did exit out of the shell and start a new one, but I am not sure if that was the fix or not as I had started fresh ones before. Anyway, thanks for the help!

**Reply by carolynvs** [reply]
Oh weird! I am not sure what happened. If you run into it again down the road and have more ideas on how to reproduce it, let us know. But I'm glad that's working for you now.
